### ISSUE TITLE
Change blues to be accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3701]
Splitting this out from #3700 

$blue-medium-500 (#00a0d2) is used in various places (mainly for links but also for other stuff) and it should be checked carefully as it has a way too low contrast ratio against white.

For example, it's used for links in the editor content, and it has a contrast that's definitely too low especially when the link is highlighted:

<img width="360" alt="screen shot 2017-11-28 at 17 16 53" src="https://user-images.githubusercontent.com/1682452/33330865-4fe9a940-d460-11e7-8d97-0e4954d2f3fa.png">

In this case, the relevant rule in the built CSS is:
```
.blocks-editable__tinymce a {
    color: #00a0d2;
}
```

against a white background, this gives a color contrast ratio of 3.02:1. It should be at least 4.5:1.

Question: should themes be able to customize the links color?

**Comment by karmatosed** [comment]
I think first up lets find a blue that is used and is acceptable contrast. Then we can change all links to that. I feel that's the first step here.

As far as themes changing the link, that feels like something which could easily create inaccessible experiences in the editor. Let's just for now have that the color is accessible.
